graph/cycle_check_directed_graph.cpp

```diff
@@ -65,8 +65,8 @@ class Graph {
      * @param vertices specify the number of vertices the graph would contain.
      * @param adjList is the adjacency list representation of graph.
      */
-    Graph(unsigned int vertices, AdjList const& adjList)
-        : m_vertices(vertices), m_adjList(adjList) {}
+    Graph(unsigned int vertices, AdjList adjList)
```

Comment:
did this line revert a good code?

Reply:
This is actually what the linter suggests:
```
/Users/filip/github/C-Plus-Plus/graph/cycle_check_directed_graph.cpp:68:34: warning: pass by value and use std::move [modernize-pass-by-value]
    Graph(unsigned int vertices, AdjList const& adjList)
                                 ^~~~~~~~~~~~~~
                                 AdjList
```

Reply:
if I understand correctly, the input parameter is copied regardless, and therefore it's encouraged to copy it directly in the parameter passing since the compiler can better optimise

Reply:
Got it. [Reference to more on this](https://stackoverflow.com/questions/26261007/why-is-value-taking-setter-member-functions-not-recommended-in-herb-sutters-cpp)